### PR TITLE
feat: add usePromise/useFetch hook

### PR DESCRIPTION
**Pull request by yuanyan**
*Before* submitting a pull request, please make sure the following is done...

1. Fork the repo and create your branch from `master`.
2. If you've added code that should be tested, add tests!
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).
5. Make sure your code lints (`npm run lint`) - we've done our best to make sure these rules match our internal linting guidelines.

### CODE REVIEW

File: packages/rax-use-fetch/src/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { useMemo } from 'rax';
+import usePromise from 'rax-use-promise';
+
+export default function useFetch(input, init, responseType = 'json') {
+  return usePromise(
+    useMemo(
```
Comment (wssgcg1213) [inline packages/rax-use-fetch/src/index.js]: use memo should be used in performance optim, not a universal logic.    
Consider that API returns current time or a real-time number of sales.

File: packages/rax-use-promise/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+```jsx
+import { createElement, useMemo } from 'rax';
+import usePromise from 'rax-use-promise';
+
+const fetchData = () => fetch('https://httpbin.org/get').then(res => res.json());
+
+function Example() {
+  const [data, error] = usePromise(useMemo(fetchData));
+  if (erro) {
```
Comment (yongningfu) [inline packages/rax-use-promise/README.md]: error

File: packages/rax-use-promise/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "rax-use-promise",
+  "version": "1.0.0-beta.0",
+  "description": "Raxx usePromise hook",
```
Comment (yongningfu) [inline packages/rax-use-promise/package.json]: Rax

File: packages/rax-use-fetch/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# rax-use-fetch
+
+```jsx
+import { createElement, useMemo } from 'rax';
+import usePromise from 'rax-use-promise';
+
+function Example() {
+  const fetchData = useMemo(() => fetch('https://httpbin.org/get').then(res => res.json()), []);
+  const [data, error] = usePromise(fetchData);
+  if (erro) {
```
Comment (yongningfu) [inline packages/rax-use-fetch/README.md]: error

File: README.md [hunk 5]
```diff
@@ -347,6 +347,46 @@ render(<Hello name="world" />, document.body, { driver: DomDriver });
...
+
+#### Asynchronous Operation 
+```jsx
+import { createElement, useMemo } from 'rax';
+import usePromise from 'rax-use-promise';
+
+function Example() {
+  const fetchData = useMemo(() => fetch('https://httpbin.org/get').then(res => res.json()), []);
+  const [data, error] = usePromise(fetchData);
+  if (erro) {
```
Comment (yongningfu) [inline README.md]: error

**Pull request merged**